docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,598 @@
+| `int`      | `"123"`                 | :material-check:      | :material-close: | Convert |
+| `int`      | `123.0`                 | :material-check:      | :material-close: | Convert |
+| `int`      | `123.1`                 | :material-check:      | :material-check: | Error   |
+| `date`     | `"2020-01-01"`          | :material-check:      | :material-close: | Convert |
+| `date`     | `"2020-01-01T12:00:00"` | :material-check:      | :material-check: | Error   |
+| `int`      | `b"1"`                  | :material-close:      | :material-close: | Error   |
+
+(For the last case converting `bytes` to an `int` could reasonably mean `int(bytes_data.decode())` or 
+`int.from_bytes(b'1', 'big')`, hence an error)
+
+In addition to the general rule, we'll provide a conversion table which defines exactly what data will be allowed 
+to which field types. See [the table below](#conversion-table) for a start on this.
+
+### Built in JSON support :thumbsup:
+
+pydantic-core can parse JSON directly into a model or output type, this both improves performance and avoids
+issue with strictness - e.g. if you have a strict model with a `datetime` field, the input must be a 
+`datetime` object, but clearly that makes no sense when parsing JSON which has no `datatime` type.
+Same with `bytes` and many other types.
+
+Pydantic V2 will therefore allow some conversion when validating JSON directly, even in strict mode 
+(e.g. `ISO8601 string -> datetime`, `str -> bytes`) even though this would not be allowed when validating
+a python object.
+
+In future direct validation of JSON will also allow:
+
+* parsing in a separate thread while starting validation in the main thread
+* line numbers from JSON to be included in the validation errors
+
+!!! note
+    Pydantic has always had special support for JSON, that is not going to change.
+
+    While in theory other formats
+    could be specifically supported, the overheads are significant and I don't think there's another format that's
+    used widely enough to be worth specific logic. Other formats can be parsed to python then validated, similarly
+    when serialising, data can be exported to a python object, then serialised, 
+    see [below](#improvements-to-dumpingserializationexport).
+
+### Validation without a Model :thumbsup:
+
+In pydantic V1 the core of all validation was a pydantic model, this led to significant overheads and complexity
+when the output data type was not a model.
+
+pydantic-core operates on a tree of validators with no "model" type required at the base of the tree.
+It can therefore validate a single `string` or `datetime` value, a `TypeDict` or `Model` equally easily.
+
+This feature will provide significant addition performance improvements in scenarios like:
+
+* adding validation to `dataclasses`
+* validating URL arguments, query strings, headers, etc. in FastAPI
+* adding validation to `TypedDict`
+* function argument validation
```

Comment:
I would have added "adding validation to your custom classes, decorators...". Thing that has been tried many times with pydantic models with custom metaclasses or whatever (see issues)